CRM/Utils/Date.php

```diff
@@ -800,17 +800,20 @@ public static function getRange($startDate, $endDate) {
    * @return array
    *   start date, end date
    */
-  public static function getFromTo($relative, $from, $to) {
+  public static function getFromTo($relative, $from, $to, $fromTime = NULL, $toTime = NULL) {
+    if (empty($toTime)) {
+      $toTime = '235959';
+    }
     if ($relative) {
-      list($term, $unit) = explode('.', $relative);
+      list($term, $unit) = CRM_Utils_System::explode('.', $relative, 2);
       $dateRange = self::relativeToAbsolute($term, $unit);
-      $from = $dateRange['from'];
+      $from = substr($dateRange['from'], 0, 8);
```

Comment:
I wonder if there could be any existing cases where it is working 'because' the string is passed through uncut here - looks like it would work :-(

Reply:
I did some checks & it seems existing usages of this function are mostly in the advanced search & they don't appear to offer time granularity so it's probably fine / safe

Reply:
I did the same check - I actually considered removing the `$fromTime` and `$toTime` from the function signature since this function *never* returns time-level granularity.  However, since my goal is to insert a hook here, I kept them in, in case someone wants to write an extension that provides it.

Reply:
OK - well I guess we both checked & didn't find something it's probably ok

Reply:
I like the fact this leaves a path to adding 'last one hour' as a filter too